vsintegration/src/vs/FsPkgs/FSharp.Project/Common.Source.CSharp/Project/HierarchyNode.cs

```diff
@@ -3301,72 +3300,61 @@ string newTargetFrameworkMoniker
             projectMgr.BuildProject.SetProperty(ProjectFileConstants.TargetFrameworkVersion, HierarchyNode.GetFrameworkVersionString(frameworkName));
             projectMgr.BuildProject.SetProperty(ProjectFileConstants.TargetFrameworkProfile, frameworkName.Profile);
 
+            string targetFSharpCoreVersion = projectMgr.BuildProject.GetPropertyValue(ProjectFileConstants.TargetFSharpCoreVersion);
+            bool autoGenerateBindingRedirects;
+            bool.TryParse(projectMgr.BuildProject.GetPropertyValue("AutoGenerateBindingRedirects"), out autoGenerateBindingRedirects);
+
 #if FX_ATLEAST_45
-            // update FSharp.Core only if we are addressing '.NETFramework' as opposed to e.g. Silverlight or Portable
-            if (frameworkName.Identifier == ".NETFramework")
```

Comment:
This seems suspect -- code below only handles desktop versioning (new assigned version is only 2.3.0.0, 4.3.0.0 or 4.4.0.0), but removing this conditional means code will need to handle assigning new portable versions too.